packages/vector_graphics_compiler/lib/src/svg/masking_optimizer.dart

```diff
@@ -0,0 +1,362 @@
+    if (node is ResolvedMaskNode) {
+      data = node.child.accept(this, data);
+    }
+    return data;
+  }
+
+  @override
+  _Result visitParentNode(ParentNode parentNode, Node data) {
+    final List<Node> newChildren = <Node>[];
+    bool deleteMaskNode = true;
+
+    for (Node child in parentNode.children) {
+      final _Result childResult = child.accept(this, parentNode);
+      newChildren.add(childResult.node);
+      if (childResult.deleteMaskNode == false) {
+        deleteMaskNode = false;
+      }
+    }
+
+    final ParentNode newParentNode = ParentNode(parentNode.attributes,
+        precalculatedTransform: parentNode.transform, children: newChildren);
+
+    final _Result _result = _Result(newParentNode);
+
+    _result.deleteMaskNode = deleteMaskNode;
+
+    return _result;
+  }
+
+  @override
+  _Result visitMaskNode(MaskNode maskNode, Node data) {
+    final _Result _result = _Result(maskNode);
+
+    return _result;
+  }
+
+  @override
+  _Result visitPathNode(PathNode pathNode, Node data) {
+    final _Result _result = _Result(pathNode);
+    return _result;
+  }
+
+  @override
+  _Result visitResolvedMaskNode(ResolvedMaskNode maskNode, void data) {
+    _Result _result = _Result(maskNode);
+    final ResolvedPathNode? singleMaskPathNode = getSingleChild(maskNode.mask);
+
+    if (singleMaskPathNode != null) {
+      masksToApply.add(singleMaskPathNode);
+      final _Result childResult = maskNode.child.accept(this, maskNode);
+      masksToApply.removeLast();
+
+      if (childResult.deleteMaskNode == true) {
+        _result = _Result(childResult.node);
+      } else {
+        final ResolvedMaskNode newMaskNode = ResolvedMaskNode(
+            child: childResult.node,
+            mask: maskNode.mask,
+            blendMode: maskNode.blendMode);
+        _result = _Result(newMaskNode);
+      }
+    } else {
+      final _Result childResult = maskNode.child.accept(this, maskNode);
+      final ResolvedMaskNode newMaskNode = ResolvedMaskNode(
+          child: childResult.node,
+          mask: maskNode.mask,
+          blendMode: maskNode.blendMode);
+      _result = _Result(newMaskNode);
+    }
+
+    return _result;
+  }
+
+  @override
+  _Result visitResolvedClipNode(ResolvedClipNode clipNode, Node data) {
+    final _Result childResult = clipNode.child.accept(this, clipNode);
+    final ResolvedClipNode newClipNode =
+        ResolvedClipNode(clips: clipNode.clips, child: childResult.node);
+    final _Result _result = _Result(newClipNode);
+    _result.children.add(childResult.node);
+
+    return _result;
+  }
+
+  @override
+  _Result visitResolvedPath(ResolvedPathNode pathNode, Node data) {
+    _Result _result = _Result(pathNode);
+    bool hasStrokeWidth = false;
+    bool deleteMaskNode = true;
+
+    if (pathNode.paint.stroke != null) {
+      if (pathNode.paint.stroke!.width != null) {
+        hasStrokeWidth = true;
+        _result.deleteMaskNode = false;
+      }
+    }
```

Comment:
Since there is no code that runs between the first and second condition, consider combining them into a single check using a null aware access:

```dart
if (pathNode.paint.stroke?.width != null) {
  hasStrokeWidth = true;
  _result.deleteMaskNode = false;
}
```

Reply:
fixed.